### PR TITLE
overlord,overlord/snapstate: have UpdateMany retire/enable auto-aliases even without new revision

### PR DESCRIPTION
**Pull request by pedronis**
This makes it so UpdateMany considers changes to auto-aliases from the snap-declaration even if there's no new updated revision.

Also tries to deal gracefully with auto-aliases transfers (even if likely rare) by retiring transfer sources' auto-aliases first.

TODO: need  most of same logic to happen also just for Update (done in #2606)

### CODE REVIEW

**Comment by pedronis** [comment]
based on #2602

File: overlord/snapstate/backend_test.go [hunk 1]
```diff
@@ -148,71 +148,74 @@ func (f *fakeStore) ListRefresh(cands []*store.RefreshCandidate, _ *auth.UserSta
...
 		}
-	}
 
-	f.fakeBackend.ops = append(f.fakeBackend.ops, fakeOp{op: "storesvc-list-refresh", cand: *cand, revno: hit})
+		f.fakeBackend.ops = append(f.fakeBackend.ops, fakeOp{op: "storesvc-list-refresh", cand: *cand, revno: hit})
 
-	if hit.Unset() {
-		return nil, nil
+		if !hit.Unset() {
+
```
Comment (mvo5) [inline overlord/snapstate/backend_test.go]: Is the extra newline here on purpose?

Comment (pedronis) [inline overlord/snapstate/backend_test.go]: no

**Comment by mvo5** [comment]
Looks fine, my comment #2606 (comment) actually applies here but we can do this in a followup, it looks like we can probably extract a helper but its fine as it is.

**Pull request merged**